lib/ruby_lsp/listeners/code_lens.rb

```diff
@@ -181,18 +181,30 @@ def add_test_code_lens(node, name:, command:, kind:)
         )
       end
 
-      sig { params(group_name: String, method_name: T.nilable(String)).returns(String) }
-      def generate_test_command(group_name:, method_name: nil)
+      sig do
+        params(
+          group_stack: T::Array[String],
+          spec_name: T.nilable(String),
+          method_name: T.nilable(String),
+        ).returns(String)
+      end
+      def generate_test_command(group_stack: [], spec_name: nil, method_name: nil)
         command = BASE_COMMAND + T.must(@path)
 
         case DependencyDetector.instance.detected_test_library
         when "minitest"
           command += if method_name
-            " --name " + "/#{Shellwords.escape(group_name + "#" + method_name)}/"
+            # Direct match only
+            " --name " + Shellwords.escape(group_stack.join("::") + "#" + method_name)
+          elsif spec_name
+            " --name " + "/#{Shellwords.escape(spec_name)}/"
           else
-            " --name " + "/#{Shellwords.escape(group_name)}/"
+            # Match all test methods that begin with the class path.
+            " --name " + "\"/\\A#{Shellwords.escape(group_stack.join("::"))}/\""
```

Comment:
Nitpick.
```suggestion
            " --name " + "\"/^#{Shellwords.escape(group_stack.join("::"))}/\""
```

Reply:
I tried using `$` and was wondering why it wasn't working 🤦